driver-core/src/main/com/mongodb/internal/connection/DefaultServer.java

```diff
@@ -132,6 +138,24 @@ public void invalidate() {
         }
     }
 
+    @Override
+    public void invalidate(final Throwable t) {
+        if ((t instanceof MongoSocketException && !(t instanceof MongoSocketReadTimeoutException))) {
+            invalidate();
+        } else if (t instanceof MongoNotPrimaryException || t instanceof MongoNodeIsRecoveringException) {
+            if (description.getMaxWireVersion() < FOUR_DOT_TWO_WIRE_VERSION) {
+                invalidate();
+            } else if (SHUTDOWN_CODES.contains(((MongoCommandException) t).getErrorCode())) {
+                invalidate();
+            } else {
```

Comment:
Let's protect this block (or the whole method) with an isClosed() check, to match the other invalidate method.

Reply:
Ack - will do